File: src/ssr_pkg/src/chao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <qq_msgs/Carry.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "chao_node");
    printf("this is chao_node~\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<qq_msgs::Carry>("chao_topic", 10);

    ros::Rate loop_rate(10);

    while (ros::ok())
    {
        qq_msgs::Carry msg;
        msg.grade = "王者";
        msg.star = 50;
        msg.data = "chao_node_publisher...";
        pub.publish(msg);
        loop_rate.sleep();
    }
    
    return 0;
}

// int main(int argc, char *argv[])
// {
//     ros::init(argc, argv, "chao_node");
//     printf("this is chao_node~\n");

//     ros::NodeHandle nh;
//     ros::Publisher pub = nh.advertise<std_msgs::String>("chao_topic", 10);

//     ros::Rate loop_rate(10);

//     while (ros::ok())
//     {
//         std_msgs::String msg;
//         msg.data = "chao_node_publisher...";
//         pub.publish(msg);
//         loop_rate.sleep();
//     }
    
//     return 0;
// }

File: src/ssr_pkg/scripts/chao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry

if __name__ == "__main__":
    rospy.init_node("chao_node")
    rospy.loginfo("this is py_chao_node~")

    pub = rospy.Publisher("chao_topic", Carry, queue_size=10)

    loop_rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        msg = Carry()
        msg.grade = "王者"
        msg.star = 50
        msg.data = "py_chao_node_publisher..."
        pub.publish(msg)

        loop_rate.sleep()

# if __name__ == "__main__":
#     rospy.init_node("chao_node")
#     rospy.loginfo("this is py_chao_node~")

#     pub = rospy.Publisher("chao_topic", String, queue_size=10)

#     loop_rate = rospy.Rate(10)

#     while not rospy.is_shutdown():
#         msg = String()
#         msg.data = "py_chao_node_publisher..."
#         pub.publish(msg)

#         loop_rate.sleep()

